lib/sla.js

```diff
@@ -189,96 +243,276 @@ function originateChannel(client, inbound, bridge, trunkData) {
+          trunkData[0], app: 'sla',
+          appArgs: 'dialed',
+          timeout: 10})
+          .catch(function(err) {
+            defer.reject(err);
+          });
+        updateState(client, bridge.name, 'RINGING');
+        channelToDial.once('StasisStart', dialedEnteredStasis);
+        channelToDial.once('ChannelDestroyed', dialedChannelHungup);
+        outbound.removeListener('ChannelHangupRequest', outboundHungupEarly);
+        outbound.once('ChannelHangupRequest', outboundHungup);
+        break;
+    default:
+      toDial += digit;
+      break;
+    }
+  }
+  var toDial = '';
+  var callInProgress = false;
+  var channelToDial;
+  var defer = Q.defer();
+  getState(client, bridge.name).then(function (currentState) {
+    if (currentState === 'INUSE' || currentState === 'RINGING' ||
+      currentState === 'BUSY') {
+      callInProgress = true;
+    }
+    if (!trunkData[0]) {
+      defer.reject(new customError.CustomError('NoTrunks', 'There are no ' +
+         'trunks with which to outbound dial'));
+    } else if (!callInProgress) {
+      var playback = client.Playback();
+      outbound.play({media: 'sound:pls-entr-num-uwish2-call'}, playback);
+      updateState(client, bridge.name, 'BUSY');
+      outbound.on('ChannelDtmfReceived', dtmfReceived);
+      outbound.once('ChannelHangupRequest', outboundHungupEarly);
+    } else {
+      defer.reject(new customError.CustomError('ExtensionBusy',
+          'Call in progress'));
+    }
+  });
   return defer.promise;
 }
 
 /** 
- * Utility function for checking if the bridgeName exists in the configuration
- *   file (is valid).
- * @param {String} configurationFile - the path and filename associated with the
- *   cofiguration file 
- * @param {String} extension - the name of the extension (and trunk so far)
- *   to be created/used
+ * Utility function for checking if a channel is a member of an extension
+ * @param {String} extension - the extension in question
+ * @param {String} channel - the channel in question
  * @return {Q} - Q promise object
  */
-function getTrunkData (configurationFile, extension) {
-  return dal.getSharedExtension(configurationFile, extension)
-    .then(function(sharedExtension) {
-      return sharedExtension.getTrunk(extension).then(function(trunk) {
-      return Q.resolve(trunk);
-    })
-    .catch(function(invalid) {
-      // This will only be called if trunk and extension are not the same
-      // Shouldn't be right now
-      return Q.reject(new customError.CustomError('InvalidTrunkSpecification',
-          invalid));
+function isStation (extension, channel) {
+  return extension.getAllStations().then(function(stations) {
+    var isStation = false;
+    stations.some(function(station) {
+      var re = regexp().find(station).toRegExp();
+      // First conditional is for regular SIP users like SIP/phone1
+      if (re.test(channel.name)) {
+        isStation = true;
+        return true; 
```

Comment:
Why return the boolean value true? Calling Return; will signal success without having to pass that value. Error conditions should be signaled with throw instead.